### PR TITLE
Write init arguments (with defaults) to yaml files upon initialisation.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: kit/hydra/relay.py [hunk 1]
```diff
@@ -41,6 +43,37 @@ def _camel_to_snake(name: str) -> str:
     return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
 
 
+def _to_yaml_value(default: Any, *, indent_level: int = 0) -> str | None:
+    str_ = None
+    if default is None:
+        str_ = "null"
+    elif isinstance(default, str):
+        str_ = default
```
Comment (thomkeh) [inline kit/hydra/relay.py:51]: `str` needs quotes in yaml if it contains a colon (`:`); maybe not a common case though

Comment (thomkeh) [reply kit/hydra/relay.py:51]: multiline strings also need special treatment

Comment (MylesBartlett) [reply kit/hydra/relay.py:51]: Right; I may as well add the former. The latter I can't be bothered to handle at the moment and I can't imagine there being many use-cases for it.

Comment (thomkeh) [reply kit/hydra/relay.py:51]: there are some other edge cases apparently: https://stackoverflow.com/a/22235064

for example, if the string could be interpreted as a number, like "10"

Comment (MylesBartlett) [reply kit/hydra/relay.py]: I was just reading this very post.

Comment (MylesBartlett) [reply kit/hydra/relay.py]: Anyway, the current converter-function should be robust for now -- should someone encounters some edge-cases later "in the wild", we can address them then.

**Pull request merged**